tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContent.scala

```diff
@@ -0,0 +1,82 @@
+package com.linagora.tmail.encrypted
+
+import com.google.common.io.ByteSource
+import com.linagora.tmail.pgp.Encrypter
+import org.apache.james.jmap.api.model.Preview
+import org.apache.james.mailbox.model.{MessageId, ParsedAttachment}
+
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicInteger
+import scala.jdk.OptionConverters._
+
+class EncryptedEmailContentFactory(encrypter: Encrypter) {
+
+  def encrypt(clearEmailContent: ClearEmailContent, messageId: MessageId): EncryptedEmailContent =
+    EncryptedEmailContent(encryptedPreview = encryptPreview(clearEmailContent.preview),
+      encryptedHtml = encrypt(clearEmailContent.html),
+      hasAttachment = clearEmailContent.hasAttachment,
+      encryptedAttachmentMetadata = encryptAttachmentMetadata(clearEmailContent.attachments, messageId),
+      encryptedAttachmentContents = encryptAttachmentContent(clearEmailContent.attachments))
+
+  def encryptPreview(preview: Preview): String =
+    encrypt(preview.getValue)
+
+  def encryptAttachmentMetadata(parsedAttachments: List[ParsedAttachment], messageId: MessageId): Option[String] = {
+    if (parsedAttachments.isEmpty) return None
+    Some(encrypt(AttachmentMetaDataSerializer
+      .serializeList(AttachmentMetadata.fromJava(parsedAttachments, messageId))
+      .toString()))
+  }
+
+  def encryptAttachmentContent(parsedAttachments: List[ParsedAttachment]): List[String] = {
+    if (parsedAttachments.isEmpty) return List.empty
+    parsedAttachments.map(parsedAttachment => encrypt(parsedAttachment.getContent))
+  }
+
+  def encrypt(byteSource: ByteSource): String = {
+    val stream: ByteArrayOutputStream = new ByteArrayOutputStream
+    encrypter.encrypt(byteSource, stream)
+    new String(stream.toByteArray, StandardCharsets.UTF_8)
+  }
+
+  def encrypt(value: String): String = {
+    val stream: ByteArrayOutputStream = new ByteArrayOutputStream
+    encrypter.encrypt(ByteSource.wrap(value.getBytes(StandardCharsets.UTF_8)), stream)
```

Comment:
you can just call `encrypt(byteSource: ByteSource)` with your string converted into a ByteSource

Reply:
Sorry, I don't understand

Reply:
```
def encrypt(value: String): String =
  encrypt(ByteSource.wrap(value.getBytes(StandardCharsets.UTF_8)))
```